File: test/api/lotsa_spec.js
var frisby = require('frisby');
var auth = require("./auth_data");

var URL = 'http://localhost:8080/api/';

frisby.globalSetup({ // globalSetup is for ALL requests
  request: {
    headers: { 'Authorization': 'Bearer ' + auth.TOKEN }
  }
});


// create a bunch of todos

var NUM_TODOS = 50;
for (var i = 0; i < NUM_TODOS; i++) {
		
	var rando = Math.random().toString(36).substring(7);
	var todo_name = "todo " + rando;

	frisby.create('POST new todo')
	  .post(URL + 'todo', {
	        name: todo_name,
	        description: "descript: " + rando
	    }, {json: true})
	  .expectStatus(200)
	  .expectJSONTypes({
	    result: String,
	    todo: Object,
	  })
	  .expectJSON({
	    result: "success!"  
	  })
	.toss();

};


// test paging here




// cleanup

frisby.create('GET all todos')
  .get(URL + 'todo')
  .expectStatus(200)
  // .expectJSON([])
  .afterJSON(function(todos) {
    // Use data from previous result in next test

    expect(todos.length).toEqual(NUM_TODOS);

    for (var i = 0; i < todos.length; i++) {
      var todo = todos[i];

      // console.log(todo);

      frisby.create('Delete todo ' + i)
        .delete(URL + 'todo/' + todo.id)
        .expectStatus(200)
      .toss();
      
    };
  })
.toss();

File: test/api/lotsa_spec.go
package main

import (
	"fmt"
	"os"

	"github.com/bitly/go-simplejson"
	"github.com/verdverm/frisby"
)

var (
	URL       = "http://localhost:8080/api/"
	UUID      = ""
	TOKEN     = ""
	NUM_TODOS = 50
)

func main() {
	UUID = os.Getenv("UUID")
	TOKEN = os.Getenv("TOKEN")

	new_todo := map[string]string{
		"name":        "lotsatodo #",
		"description": "this is my first todo posted from frisby!",
	}
	var todo_ids []string

	POST := frisby.Create("POST lotsa todo collector")
	DELETE := frisby.Create("DELETE lotsa todo collector")

	for i := 0; i < NUM_TODOS; i++ {
		new_todo["name"] = "lotsatodo " + fmt.Sprint(i)
		todo_id_int := 0
		frisby.Create("POST lotsa todo: "+fmt.Sprint(i)).
			Post(URL+"todo").
			SetHeader("Authorization", "Bearer "+TOKEN).
			SetJson(new_todo).
			Send().
			ExpectStatus(200).
			ExpectJson("result", "success!").
			AfterJson(func(F *frisby.Frisby, json *simplejson.Json, err error) {

			todo_id_json := json.GetPath("todo", "id")
			var cerr error
			todo_id_int, cerr = todo_id_json.Int()
			if cerr != nil {
				POST.AddError(cerr.Error())
			}
			tid := fmt.Sprint(todo_id_int)
			todo_ids = append(todo_ids, tid)

		})
	}
	POST.PrintReport()

	for i := 0; i < NUM_TODOS; i++ {
		todo_id := todo_ids[i]
		errs := frisby.Create("DELETE lotsa todo: "+todo_id).
			Delete(URL+"todo/"+todo_id).
			SetHeader("Authorization", "Bearer "+TOKEN).
			Send().
			ExpectStatus(200).
			ExpectJson("result", "success!").
			Errors()

		for _, e := range errs {
			DELETE.AddError(e.Error())
		}
	}
	DELETE.PrintReport()
}
